microservices/submission-service/src/replay-validation/replay-validation.service.spec.ts

```diff
@@ -0,0 +1,747 @@
+                status: ResponseStatus.SUCCESS,
+                data: [
+                    {
+                        status: ResponseStatus.SUCCESS,
+                        success: true,
+                        data: {
+                            id: 1,
+                            userId: 1,
+                            skillGroupId: 5,
+                            franchise: {
+                                name: "Frogs",
+                            },
+                        },
+                    },
+                ],
+            };
+        });
+
+        when(minioService.get(anything(), anything())).thenCall(async () => {
+            return Readable.from(JSON.stringify({data: rawResponse}));
+            // return rawResponse;
+        });
+
+        const mmsInstance = instance(matchmakingService);
+        const csInstance = instance(coreService);
+        const msInstance = instance(minioService);
+        service = new ReplayValidationService(csInstance, mmsInstance, msInstance);
+    });
+
+    it("should be defined", () => {
+        expect(service).toBeDefined();
+    });
+
+    describe("Sort IDs:", () => {
+        it("Should sort properly", () => {
+            const inputGames = [
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+                [
+                    [2, 1],
+                    [3, 4],
+                ],
+                [
+                    [4, 3],
+                    [2, 1],
+                ],
+            ];
+            const expectedOutput = [
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+            ];
+            expect(sortIds(inputGames)).toStrictEqual(expectedOutput);
+        });
+    });
+
+    describe("Check parent method branches properly:", () => {
+        it("Should validate as a scrim", async () => {
+            const scrimResponseObject = {
+                valid: true,
+                errors: [
+                    {
+                        error: "This is a scrim",
+                    },
+                ],
+            };
+            service.validateScrimSubmission = async _ => {
+                return scrimResponseObject;
+            };
+            expect(await service.validate(testSubmission)).toStrictEqual(scrimResponseObject);
+        });
+
+        it("Should validate as a match", async () => {
+            const matchResponseObject = {
+                valid: true,
+                errors: [
+                    {
+                        error: "This is a match",
+                    },
+                ],
+            };
+            service.validateMatchSubmission = async _ => {
+                return matchResponseObject;
+            };
```

Comment:
Do we want to fully override this function, or replace it with a mock?

Reply:
This way is much cleaner, so I'm leaving it as it for now.